xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2921,7 +2921,7 @@ public Locale getLocalePreference(XWikiContext context)
         // in use.
         try {
             String language = Util.normalizeLanguage(context.getRequest().getParameter("language"));
-            if (StringUtils.isNotEmpty(language)) {
+            if (language != null) {
```

Comment:
I don't think this change is needed anymore.

Reply:
I don't think it fixes anything, but semantically it looks more correct since we might have an empty language parameter which is correct.
Basically it might avoid the first reaction you had when seeing that :) WDYT?

Reply:
My point was more that it was not really linked to this issue but sure.

Reply:
Right. So if you're ok to merge with it I'll keep the change.

Reply:
As you prefer.